summingbird-storm-test/src/test/scala/com/twitter/summingbird/storm/StormLaws.scala

```diff
@@ -173,177 +106,159 @@ class StormLaws extends WordSpec {
     val fn2 = sample[(Int) => List[(Int, Int)]]
     val fn3 = sample[(Int) => List[(Int, Int)]]
 
-    val (store1Id, store1) = genStore
-
     val preJoinFn = sample[(Int) => (Int, Int)]
     val postJoinFn = sample[((Int, (Int, Option[Int]))) => List[(Int, Int)]]
 
     val serviceFn = sample[Int => Option[Int]]
-    val service = ReadableServiceFactory[Int, Int](() => ReadableStore.fromFn(serviceFn))
-
-    val tail = TestGraphs.realJoinTestJob[Storm, Int, Int, Int, Int, Int, Int, Int, Int, Int](source1, source2, source3, source4,
-      service, store1, fn1, fn2, fn3, preJoinFn, postJoinFn)
-
-    assert(OnlinePlan(tail).nodes.size < 10)
-    StormTestRun(tail)
-
-    val scalaA = TestGraphs.realJoinTestJobInScala(original1, original2, original3, original4,
-      serviceFn, fn1, fn2, fn3, preJoinFn, postJoinFn)
 
-    val store1Map = TestStore[Int, Int](store1Id).get.toScala
-    assertEquiv[Map[Int, Int]](
-      scalaA,
-      store1Map
-    )
+    testStormEqualToMemory(new ProducerCreator {
+      override def apply[P <: Platform[P]](ctx: CreatorCtx[P]): TailProducer[P, Any] =
+        TestGraphs.realJoinTestJob[P, Int, Int, Int, Int, Int, Int, Int, Int, Int](
+          ctx.source(original1), ctx.source(original2), ctx.source(original3), ctx.source(original4),
+          ctx.service(serviceFn), ctx.store("store"), fn1, fn2, fn3, preJoinFn, postJoinFn)
+    })
   }
 
-  def assertEquiv[T](expected: T, returned: T)(implicit equiv: Equiv[T]): Unit = {
-    assert(equiv.equiv(expected, returned), (expected.toString, returned.toString))
+  "StormPlatform should be able to handle AlsoProducer with Summer and FlatMap in different branches" in {
+    val original = sample[List[(Int, Int)]]
+    val branchFlatMap = sample[((Int, Int)) => List[(Int, Int)]]
+
+    testStormEqualToMemory(new ProducerCreator {
+      override def apply[P <: Platform[P]](ctx: CreatorCtx[P]): TailProducer[P, Any] = {
+        val source = ctx.source(original)
+        source.sumByKey(ctx.store("store1")).also(
+          source.flatMap(branchFlatMap).sumByKey(ctx.store("store2"))
+        )
+      }
+    })
+
+//    Fails, see https://github.com/twitter/summingbird/issues/725 for details.
+//    testStormEqualToMemory(new ProducerCreator {
+//      override def apply[P <: Platform[P]](ctx: CreatorCtx[P]): TailProducer[P, Any] = {
+//        val source = ctx.source(original).flatMap(e => List(e))
+//        source.sumByKey(ctx.store("store1")).also(
+//          source.flatMap(branchFlatMap).sumByKey(ctx.store("store2"))
+//        )
+//      }
+//    })
+
+//    Workaround
+    testStormEqualToMemory(new ProducerCreator {
+      override def apply[P <: Platform[P]](ctx: CreatorCtx[P]): TailProducer[P, Any] = {
+        val source = ctx.source(original).flatMap(e => List(e))
+        source.map(identity).sumByKey(ctx.store("store1")).also(
+          source.flatMap(branchFlatMap).sumByKey(ctx.store("store2"))
+        )
+      }
+    })
+
+//    Also fails.
```

Comment:
I think we should make issues about this and see if they should be gone, or if we can find a way to make them pass. I haven't looked at them much. They may be assuming a stronger contract than actually exists.